File: CodingSamples/Java/Runtime/GenMethodTest/Program1.java
class Program {
	
	private static String select(int choice, String first, String second) {
		if((choice % 2) == 1)
			return first;
		return second;
	}
	
	private static double select(int choice, double first, double second) {
		if((choice % 2) == 1)
			return first;
		return second;
	}

	public static void main(String[] args) {
		if(args.length > 0){
			int s = Integer.parseInt(args[0]);
			String ss = select(s, "Monday", "Friday");
			System.out.printf("Selected String = %s%n", ss);
			double sd = select(s, 23.4, 43.2);
			System.out.printf("Selected double = %s%n", sd);
			//int si = select(s, "abcd", 1234);
		}
	}
}

File: CodingSamples/DotNet/Runtime/GenMethodTest/DemoApp/Program1.cs
class Program
{
    static string Select(int index, string first, string second)
    {
        if((index % 2) == 1)
            return first;
        return second;
    }

    static double Select(int index, double first, double second)
    {
        if((index % 2) == 1)
            return first;
        return second;
    }

    static void Main(string[] args)
    {
        if(args.Length > 0)
        {
            int s = int.Parse(args[0]);
            string ss = Select(s, "Monday", "Friday");
            Console.WriteLine($"Selected string = {ss}");
            double sd = Select(s, 54.3, 67.8);
            Console.WriteLine($"Selected string = {sd}");
            //sd = Select(s, "abcd", 123.4);
        }
    }
}
